Lib/asyncio/base_events.py

```diff
@@ -647,6 +647,43 @@ def _getaddrinfo_debug(self, host, port, family, type, proto, flags):
         return await self.run_in_executor(
             None, socket.getnameinfo, sockaddr, flags)
 
+    async def sock_sendfile(self, sock, file, offset=0, count=None,
+                            *, fallback=True):
+        if self._debug and sock.gettimeout() != 0:
+            raise ValueError("the socket must be non-blocking")
+        socket._check_sendfile_params(sock, file, offset, count)
+        try:
+            await self._sock_sf_fast(sock, file, offset, count)
```

Comment:
I'd rename to `_sock_sendfile_native` and `_sock_sendfile_fallback`.   "sf" is unintuitive.